UI/win-update/updater/updater.cpp

```diff
@@ -1188,6 +1189,8 @@ static bool Update(wchar_t *cmdLine)
 			for (int i = 0; i < argc; i++) {
 				if (wcscmp(argv[i], L"Portable") == 0) {
 					bIsPortable = true;
+				} else {
+					branch = argv[i];
```

Comment:
I'm not sure if this would work as intended (not familiar with low level argument parsing). Perhaps the branch name could just be prefixed with something to be sure (e.g. `-B<branch>`)?

Reply:
I think in general there needs to be better command line parsing in the updater, it's also needed for another feature I want where OBS passes the manifest path in the command line to allow updating by non-admin users. Perhaps requiring key=value is the way to go, though for compatibility we have to consider the updater can be ran by any older OBS version :(.

Reply:
Perhaps a bit jank, but for simple `--<key>=<value>` matching couldn't we do something like this:
```cpp
wstring branch;
wstraing manifest_path;

for (int i = 0; i < argc; i++) {
    // Legacy OBS
    if (wcscmp(argv[i], L"Portable") == 0) {
        bIsPortable = true;
        break;
    }
    
    if (wcsncmp(argv[i], L"--branch=", 9) == 0) {
        branch = argv[i] + 9;
    } else if (wcsncmp(argv[i], L"--manifest=", 11) == 0) {
        manifest_path = argv[i] + 11;
    }
}
```

Reply:
That should be OK, assuming CommandLineToArgvW handles quoted paths with spaces in the name.

Reply:
Based on https://docs.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-commandlinetoargvw it should:

> CommandLineToArgvW treats whitespace outside of quotation marks as argument delimiters.

With the special note that a quotation mark immediately preceeded by a backslash is considered to be part of a file path, and not an escape sequence.

Reply:
I've implemented this in https://github.com/obsproject/obs-studio/pull/6907/commits/c9400cc5fa7e0d512777aa8686a7894e95fd1d92 now, I'm not too big a fan of what's needed to convert from a `std::string` to `std::wstring`, unfortunately I may have to go the slightly uglier route of dealing with the raw string and OBS's utilities for this.

Reply:
> warning STL4017: std::wbuffer_convert, std::wstring_convert, and the <codecvt> header (containing std::codecvt_mode, std::codecvt_utf8, std::codecvt_utf16, and std::codecvt_utf8_utf16) are deprecated in C++17. (The std::codecvt class template is NOT deprecated.) The C++ Standard doesn't provide equivalent non-deprecated functionality; consider using MultiByteToWideChar() and WideCharToMultiByte() from <Windows.h> instead. 

Well fuck me then.

Reply:
Alright, I think I have fixed it now.

Not sure how I would debug the updater itself quite yet, but I'm positive that at least will work.